### PR TITLE
Project 5: Controllers Specs

### PR DESCRIPTION
**Pull request by MoRaad97**
## Project requirements
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/required/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    it 'render a template' do
+      get('/users/4/posts')
+      expect(response).to render_template('index')
+    end
+    it 'Response Status To Be 200' do
+      get('/users/4/posts')
+      expect(response).to have_http_status(200)
+    end
+  end
+
```
Comment (danmainah) [inline spec/required/posts_controller_spec.rb]: - [x] Kindly add a test that ensures the response body includes the correct placeholder text. Also kindly ensure you do all tests for each individual view(index and show), in this case, you should have 6 tests or more.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @MoRaad97, This is @sja-thedude your code reviewer.

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that is Blog App: Controllers Spec. But for review, you selected Blog app - Views. Make sure you submit and select the same name for the project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/required/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    it 'render a template' do
+      get("/")
+      expect(response).to render_template('index')
+    end
+    it 'Response Status To Be 200' do
+      get("/")
+      expect(response).to have_http_status(200)
+    end
+  end
+
```
Comment (danmainah) [inline spec/required/users_controller_spec.rb:22]: - [x]  Kindly add a test that ensures the response body includes the correct placeholder text. Also kindly ensure you do all tests for each individual view(index and show), in this case, you should have 6 tests or more.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @MoRaad97, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by ShahierNashaat — APPROVED** [review]
Hi @MoRaad97, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**